openpype/hosts/tvpaint/api/lib.py

```diff
@@ -540,3 +540,42 @@ def get_scene_bg_color(communicator=None):
     if not data:
         return None
     return data.split(" ")
+
+
+def get_layer_opacity(layer_id, communicator=None):
+    """Return the opacity set on layer.
+    layer_id(int): id of the layer to get the opacity
+
+    Returns:
+        int: Layer Opacity (0-100).
+    """
+    output_file = tempfile.NamedTemporaryFile(
+        mode="w", prefix="a_tvp_", suffix=".txt", delete=False
+    )
+    output_file.close()
+    output_filepath = output_file.name.replace("\\", "/")
+
+    george_script_lines = [
+                # Variable containing full path to output file
+                "output_path = \"{}\"".format(output_filepath),
+                "tv_layerset {}".format(layer_id),
+                "tv_layerdensity 100",
+                "orig_opacity = result",
```

Comment:
c'est comme ça que fonctionne le george script
Lorsqu'on lance une fonction tvpaint, il return une valeur nommée toujours "result".
Ce code est inspiré du reste, si on passe par un fichier text, c'est pour eviter de rompre une certaines synch, si j'ai compris la différence en le executeGeorgeScript et excecuteGeorgeScriptThroughtFile.

Reply:
Pretty sure this would work without creating a file:
```python
# Setting the correct layer we want info
execute_george("tv_layerset {}".format(layer_id))
# Getting the layer opacity
response = execute_george("tv_layerdensity")
return int(response["result"])
```

Reply:
Yes that's what I did in a first test
I just did a similar code as "get_scene_bg_color" just the function above, to match the same logic as the other settings getter in the scene
The correct code would be:

```
# Setting the correct layer we want info
execute_george("tv_layerset {}".format(layer_id))
# Getting the layer opacity
opacity= execute_george("tv_layerdensity 100")
execute_george("tv_layerdensity {}".format(opacity))
return int(opacity)
```

But with that, we execute 3 commands, with the current  code, only one is used and it doesn't break the client websocket . I don't know which one would be the more efficient.

If you think the new one would be better let me know, I'll correct it in the next push

Reply:
no need to apply an opacity to `tv_layerdensity` and to re-set it after. without argument it just a get command, with argument it's a get and apply the desired value

Reply:
this should be working fine:
```
# Setting the correct layer we want info
execute_george("tv_layerset {}".format(layer_id))
# Getting the layer opacity
return int(execute_george("tv_layerdensity"))
```

Reply:
executing two georgescript commands is way more efficient than executing a script and pushing the result into a file, than parsing the file created

Reply:
I'll still have to do a third command, because just "tv_layerdensity" set the opacity at 0 on the layer ^^'

Reply:
From the official doc you can get the density without modifying it with `tv_layerinfo`:
![Screenshot 2024-09-03 123813](https://github.com/user-attachments/assets/5ef97fe9-2c03-40b7-8659-b7a6228b6527)

Reply:
Don 't know if it's the tvpaint11 or 12 GeorgeScript doc but worth to try it

Reply:
Careful with `tv_layerinfo` the density/opacity has a range of `[0-255]` not the classic `[0-100]`

Reply:
Ah just found out that `tv_layerinfo` is not good to use:
> It was found out that `opacity` returned from `tv_layerinfo` is always empty and is dangerous to add it to layer information.

This is from the doc of OpenPype, so yes you need a last georgescript to force the density back but 3 command is still better than creating an writting to a file if you only need to call three very simple commands